ec2/ec2test/subnets_test.go

```diff
@@ -0,0 +1,127 @@
+//
+// goamz - Go packages to interact with the Amazon Web Services.
+//
+//   https://wiki.ubuntu.com/goamz
+//
+// Copyright (c) 2015 Canonical Ltd.
+//
+
+package ec2test_test
+
+import (
+	. "gopkg.in/check.v1"
+
+	"gopkg.in/amz.v3/ec2"
+)
+
+func (s *S) TestAddSubnet(c *C) {
+	subnet, err := s.srv.AddSubnet(ec2.Subnet{})
+	c.Assert(err, ErrorMatches, "empty VPCId field")
+	c.Assert(subnet, DeepEquals, ec2.Subnet{})
+	s.assertVPCsExist(c, true)
```

Comment:
It's not clear why this assert is called here. If it's a precondition that should be true before AddSubnet and true after, then please add an additional call to assertVPCsExist before calling AddSubnet. If the call to AddSubnet *changes* something about VPCs, please add a corresponding assert before AddSubnet.